### DISCUSSION TITLE
View Source links to static line number in latest branch source instead of the version it was compiled against

### DISCUSSION THREAD

**Discussion by mikernet** [dotnet/docfx#8344] [Q&A]
For example. this is a "View Source" link that is generated:
https://github.com/Singulink/Singulink.Collections/blob/main/Source/Singulink.Collections/HashSetDictionary.cs/#L266

When really, it should be the "permalink" as follows, otherwise as the code evolves and changes, the links will no longer be valid:
https://github.com/Singulink/Singulink.Collections/blob/0555c704bdb939a21863f67de7ed68a7f6e07600/Source/Singulink.Collections/HashSetDictionary.cs#L266

Is there a way to configure the view source links to point to the permalink instead of something that can and does change?

**Comment by KalleOlaviNiemitalo** [comment]
GitHub URL seems to be constructed here: <https://github.com/dotnet/docfx/blob/02bc21e38652d9ed594d01287fa038d9a554b443/templates/common/common.js#L74>

A quick hack might be to detach HEAD and clear all the [branch-name environment variables](https://github.com/dotnet/docfx/blob/02bc21e38652d9ed594d01287fa038d9a554b443/src/Microsoft.DocAsCode.Common/Git/GitUtility.cs#L38-L47) so that DocFX [uses the commit ID rather than the branch name](https://github.com/dotnet/docfx/blob/02bc21e38652d9ed594d01287fa038d9a554b443/src/Microsoft.DocAsCode.Common/Git/GitUtility.cs#L275-L278). However, this would likely affect the "Improve this doc" links as well, and might be undesirable there.
